treeherder/webapp/api/logslice.py

```diff
@@ -58,35 +59,45 @@ def list(self, request, project, jm):
         if start_line >= end_line:
             return Response("``end_line`` must be larger than ``start_line``", 400)
 
-        if len(log) > 0:
-            try:
-                url = log[0].get("url")
-                gz_file = filesystem.get(url)
-
-                if not gz_file:
-                    handle = self.get_log_handle(url)
-                    gz_file = gzip.GzipFile(fileobj=BytesIO(handle.read()))
-                    filesystem.set(url, gz_file.fileobj)
-                else:
-                    gz_file = gzip.GzipFile(fileobj=gz_file)
-
-                lines = []
+        # @todo: remove once Bug 1139517 is addressed
+        if log_name == "buildbot_text":
+            log_name = 'builds-4h'
 
-                for i, line in enumerate(gz_file):
-                    if i < start_line:
-                        continue
-                    elif i >= end_line:
-                        break
+        # get only the log that matches the ``log_name``
+        logs = jm.get_log_references(job_id)
 
+        try:
+            log = next(iter([log for log in logs if log["name"] == log_name]))
```

Comment:
``iter([a for a in b])`` is equivalent to ``a for a in b``, so this can be written as:
``log = next(log for log in logs if log["name"] == log_name)``